src/config.rs

```diff
@@ -0,0 +1,1191 @@
+use std::{
+    collections::HashSet,
+    env::var,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{Error, Result};
+use directories::BaseDirs;
+use serde::Deserialize;
+
+pub mod limits;
+mod xml;
+
+use xml::{
+    Document, Element, PolicyContext, PolicyElement, RuleAttributes, RuleElement, TypeElement,
+};
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub enum Access {
+    Allow,
+    Deny,
+}
```

Comment:
nit: please move it below its first use.

Reply:
Do you mean move it lower, so that it's closer to its first use? Although, I guess these are hoisted in Rust, so it doesn't technically need to be above its first use... 

Would you say your preferred declaration order is public before private, related items closer than unrelated, and then alphabetic order?

Reply:
> Do you mean move it lower, so that it's closer to its first use?

No, so that the main/more important API (e.g `Config` here) comes first. A good way to ensure that is by having use before declaration/definition. Think which API will someone be most interested in when opening this file.

> Although, I guess these are hoisted in Rust, so it doesn't technically need to be above its first use...

Yeah, Rust allows us to 
 
> Would you say your preferred declaration order is public before private, related items closer than unrelated, and then alphabetic order?

Public before private but if there are two items with the same visibility, then use before definition for the reason I mentioned above. No need for Alphabetical ordering.

Reply:
Gotcha, makes sense

It's a shame there isn't some linter or auto-formatter that we can use to prevent repeats of this discussion :)